open-apparel-registry/open-apparel-registry
# Add ApiBlock Dashboard

## Overview

Adds the ApiBlock list view and the ApiBlock update view. Users can
view and update the ApiBlocks' grace_limit and grace_reason.

The list view displays active, not expired blocks in one table at the
top, and the rest of the blocks in another table. The full list of
blocks is filterable by contributor. This filter is client-side only.

Connects #1144 

## Demo

<img width="1303" alt="Screen Shot 2020-11-09 at 3 17 11 PM" src="https://user-images.githubusercontent.com/21046714/98596831-05135280-22a6-11eb-9a40-79717bce0b5c.png">
<img width="1097" alt="Screen Shot 2020-11-09 at 3 17 46 PM" src="https://user-images.githubusercontent.com/21046714/98596841-093f7000-22a6-11eb-9b49-b35ab92ad57a.png">
<img width="650" alt="Screen Shot 2020-11-09 at 3 18 36 PM" src="https://user-images.githubusercontent.com/21046714/98596847-0b093380-22a6-11eb-8839-79376508e52f.png">

## Testing Instructions

* Run `./scripts/server`
* Ensure you have at least one active and one inactive Api Block in your database. 
* Navigate to [:6543/dashboard](http://localhost:6543/dashboard).
* Click on 'View API Blocks'
* You should be navigated to the API Block list. The top list should include only active, unexpired blocks. 
* The bottom list should include all the blocks you have in your database. 
* You should be able to filter the bottom list by contributor. 
* Click on an active block in the top list. You should be taken to the update view. 
* Enter a grace limit above the actual count and request limit. Enter a grace reason. Click submit.
* The ApiBlock should be updated. 
* Return to the list by clicking the 'back to Api Blocks' button. You should see the ApiBlock was updated here as well. 
* Go to an active block's update page. Attempt to enter a grace limit below the actual count or request limit. You should see an error. 

## Checklist
